src/cocotb/regression.py

```diff
@@ -90,24 +91,62 @@
         assert False, "pytest.raises doesn't raise an exception when it fails"
 
 
-class _Test:
-    _id_count = 0  # used by the RegressionManager to sort tests in definition order
+class Test:
+    """A cocotb test in a regression.
+
+    Args:
+        func:
+            The test function object.
+
+        name:
+            The name of the test function.
+            Defaults to ``func.__qualname__`` (the dotted path to the test function in the module)
+
+        module:
+            The name of the module containing the test function.
+            Defaults to ``func.__module__`` (the name of the module containing the test function).
+
+        doc:
+            The docstring for the test.
+            Defaults to ``func.__doc__`` (the docstring of the test function).
+
+        timeout_time:
+            Simulation time duration before timeout occurs.
```

Comment:
```suggestion
            Simulation time duration before the timeout exception :exc:`~cocotb.result. SimTimeoutError` occurs.
```

Reply:
That implies it can be caught, but it can't, so I reworded it slightly.